main_stack.yaml

```diff
@@ -48,47 +48,109 @@ Resources:
     Properties:
       AvailabilityZones: !GetAZs ''
       LaunchConfigurationName: !Ref 'LaunchConfig'
-      MinSize: '0'
+      MinSize: '1'
       MaxSize: '2'
       LoadBalancerNames:
         - Fn::ImportValue: !Sub "${NetworkStackName}-MyELB"
           
 # LaunchConfig config
   LaunchConfig:
     Type: AWS::AutoScaling::LaunchConfiguration
+    Metadata:
+      AWS::CloudFormation::Authentication:
+        S3AccessCreds:
+          type: "S3"
+          role: !Ref 'RoleS3ReadAccess'
+          buckets: 
+            - "mzol-backet"
+      AWS::CloudFormation::Init:
+        install_app:
+          packages:
+            yum:
+              httpd: []
+              tomcat: []
+              java-1.8.0-openjdk: []
+          files:
+            /usr/share/tomcat/webapps/app-1.0.0.war:
+              source: "http://mzol-backet.s3.amazonaws.com/app-1.0.0.war"
+              mode: 000644
+              owner: tomcat
+              group: tomcat
+              authentication: S3AccessCreds
+            /etc/httpd/conf.d/tomcatvh.conf:
+              content: |
+                <VirtualHost *:80>
+                  ProxyPass / http://localhost:8080/app-1.0.0/app
+                  ProxyPassReverse / http://localhost:8080/
+                </VirtualHost>
+              mode: 000644
+              owner: root
+              group: root
+          services:
+            sysvinit:
+              httpd:
+                enabled: "true"
+                ensureRunning: "true"
+              tomcat:
+                enabled: "true"
+                ensureRunning: "true"
+        configSets:
+          default: [ install_app ]
     Properties:
       KeyName: !Ref 'KeyName'
       ImageId: !FindInMap [AWSRegionArch2AMI, !Ref 'AWS::Region', !FindInMap [AWSInstanceType2Arch,
           !Ref 'InstanceType', Arch]]
       InstanceType: !Ref 'InstanceType'
       SecurityGroups: [!Ref 'MySecurityGroup']
+      IamInstanceProfile: !Ref 'WebServerInstanceProfile'
       UserData:
         Fn::Base64: !Sub |
           #!/bin/bash
           yum -y update
```

Comment:
Lets write all yum commands in retry. Because we can observe process lock errors.